dgl_sparse/include/sparse/torch_headers.h

```diff
@@ -0,0 +1,29 @@
+/**
+ *  Copyright (c) 2022 by Contributors
+ * @file sparse/torch_headers.h
+ * @brief Pytorch headers used in the sparse library. Since Pytorch 2.1.0
+ * introduced a dependency on <windows.h>, we need to define NOMINMAX to avoid
+ * the conflict with std::min/std::max macros before including Pytorch headers.
+ * See more in
+ * https://stackoverflow.com/questions/4913922/possible-problems-with-nominmax-on-visual-c.
+ */
+
+#ifndef SPARSE_TORCH_HEADERS_H_
+#define SPARSE_TORCH_HEADERS_H_
+
+#ifdef _WIN32
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif  // NOMINMAX
```

Comment:
```suggestion
#ifndef NOMINMAX
#define SPARSE_TORCH_HEADERS_H_NOMINMAX
#define NOMINMAX
#endif  // NOMINMAX
```

Reply:
I didn't get it. What is the difference?

Reply:
In your implementation, #ifdef _WIN32, the NOMINMAX will be undefined regardless whether it is defined outside this header file or not. While the new one will only undefine it if the NOMINMAX by this file.

Reply:
I see. Totally make sense.